processor/src/chiplets/aux_trace/bus/hasher.rs

```diff
@@ -0,0 +1,548 @@
+use miden_air::{
+    trace::{
+        chiplets::hasher::{
+            HASH_CYCLE_LEN, LINEAR_HASH_LABEL, MP_VERIFY_LABEL, MR_UPDATE_NEW_LABEL,
+            MR_UPDATE_OLD_LABEL, NUM_ROUNDS, RETURN_HASH_LABEL, RETURN_STATE_LABEL,
+        },
+        main_trace::MainTrace,
+    },
+    RowIndex,
+};
+use vm_core::{Felt, FieldElement, ONE, ZERO};
+
+use super::{
+    get_op_label,
+    messages::{
+        ControlBlockRequestMessage, EndBlockMessage, HasherMessage, RespanBlockMessage,
+        SpanBlockMessage,
+    },
```

Comment:
I would maybe move these message structs into this file.

Reply:
done